src/nvim/terminal.c

```diff
@@ -250,12 +253,16 @@ Terminal *terminal_open(TerminalOptions opts)
   rv->sb_size = MIN(rv->sb_size, 100000);
   rv->sb_buffer = xmalloc(sizeof(ScrollbackLine *) * rv->sb_size);
 
+  if (!true_color) {
+      return rv;
+  }
+
   // Configure the color palette. Try to get the color from:
   //
   // - b:terminal_color_{NUM}
   // - g:terminal_color_{NUM}
   // - the VTerm instance
```

Comment:
the lines removed below mean that "the VTerm instance" is no longer queried for default colors. Why was that removed? And if we go with that, then this comment needs to be updated to reflect that.